README.md

```diff
@@ -9,19 +9,34 @@ Can be used as a module and from the command line.
 
 See the [CHANGELOG] for details about the latest release.
 
-Features
+# Table of contents
+1. [Features](#features)
+2. [How to install](#howToInstall)
+3. [Usage](#usage)
+    1. [CLI](#cli)
+    2. [Javascript Module](#jsModule)
+4. [json2csv Parser (Synchronous API)](#json2csvSync)
+5. [json2csv Async Parser (Streaming API)](#json2csvAsync)
+5. [json2csv Transform (Streaming API)](#json2csvTransform)
+6. [Upgrading Your Version of JSON2CSV](#upgrading)
+7. [Known Gotchas](#knownGotchas)
+8. [Contributing](#contributing)
```

Comment:
The right table of content should be:

```md
1. Features
2. How to install
3. Usage
    1. CLI
    2. Javascript Module
        1. json2csv Parser (Synchronous API)
        2. json2csv Async Parser (Streaming API)
        3. json2csv Transform (Streaming API)
9. Upgrading
10. Known Gotchas
11. Development
```

Reply:
Using an automated TOC generator I got:

```md
- [json2csv](#json2csv)
  * [How to install](#how-to-install)
  * [Usage](#usage)
    + [Command Line Interface](#command-line-interface)
  * [Javascript module](#javascript-module)
    + [Transforms](#transforms)
      - [Custom transforms](#custom-transforms)
      - [Built-in Transforms](#built-in-transforms)
        * [Unwind](#unwind)
        * [Flatten](#flatten)
    + [Formatters](#formatters)
      - [Custom Formatters](#custom-formatters)
      - [Built-in Formatters](#built-in-formatters)
        * [Default](#default)
        * [Number](#number)
        * [String](#string)
        * [String Quote Only Necessary](#string-quote-only-necessary)
        * [String Excel](#string-excel)
        * [Symbol](#symbol)
        * [Object](#object)
    + [json2csv Parser (Synchronous API)](#json2csv-parser--synchronous-api-)
    + [json2csv Async Parser (Streaming API)](#json2csv-async-parser--streaming-api-)
    + [json2csv Transform (Streaming API)](#json2csv-transform--streaming-api-)
  * [Upgrading](#upgrading)
    + [Upgrading from 5.X to 6.X](#upgrading-from-5x-to-6x)
    + [Upgrading from 4.X to 5.X](#upgrading-from-4x-to-5x)
    + [Upgrading from 3.X to 4.X](#upgrading-from-3x-to-4x)
  * [Known Gotchas](#known-gotchas)
    + [Excel support](#excel-support)
      - [Avoiding excel auto-formatting](#avoiding-excel-auto-formatting)
      - [Avoiding CSV injection](#avoiding-csv-injection)
      - [Preserving new lines](#preserving-new-lines)
      - [Unicode Support](#unicode-support)
    + [PowerShell escaping](#powershell-escaping)
  * [Development](#development)
    + [Pulling the repo](#pulling-the-repo)
    + [Building](#building)
    + [Linting & Testing](#linting---testing)
    + [Contributing changes](#contributing-changes)
  * [License](#license)
```

Reply:
Trying your branch some of the TOC links seem to work and some don't.

Reply:
Oh no! some weren't working?  I'll add the one you created with the script. Great find with that

Reply:
I didn't notice this before -Did you want me to make a TOC for the two .md files in the docs directory?

Reply:
TOCs are good.
But what I think that we really need to organize the whole docs better.